### PR TITLE
Enable flake8 for examples and tests

### PR DESCRIPTION
**Pull request by cmarqu**
Do not exclude the directories ``examples`` and ``tests`` from the flake8 config in ``setup.cfg`` anymore;
fix flake8 reports in those directories.

There is one point where I wasn't entirely sure whether that's the best fix; it is marked with
``# FIXME: can we be more specific?``, please review.

### CODE REVIEW

File: tests/test_cases/test_cocotb/test_generator_coroutines.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def test_function_not_decorated(dut):
     except TypeError as exc:
         assert "yielded" in str(exc)
         assert "scheduler can't handle" in str(exc)
-    except:
+    except Exception:  # FIXME: can we be more specific?
```
Comment (eric-wieser) [inline tests/test_cases/test_cocotb/test_generator_coroutines.py:53]: I think this is actually a bug - translating from one execution to another here isn't really useful.

What makes this a bug is the fact it's not failing if no exception is thrown, which would be spelt:
```suggestion
    else:
```

File: tests/test_cases/test_avalon_stream/test_avalon_stream.py [hunk 2]
```diff
@@ -55,7 +53,7 @@ def test_avalon_stream(dut):
     tb.backpressure.start(wave())
 
     for _ in range(20):
-        data = random.randint(0,(2^7)-1)
+        data = random.randint(0,(2 ^ 7)-1)
```
Comment (eric-wieser) [inline tests/test_cases/test_avalon_stream/test_avalon_stream.py:56]: This is meant to be an exponent, right?

```suggestion
        data = random.randint(0, (2**7)-1)
```

Comment (cmarqu) [reply tests/test_cases/test_avalon_stream/test_avalon_stream.py]: Oh, yes, likely; fixed.

File: tests/test_cases/test_cocotb/test_generator_coroutines.py (hunk 1 above)
Comment (cmarqu) [reply tests/test_cases/test_cocotb/test_generator_coroutines.py]: Done.

**Comment by eric-wieser** [comment]
Could you avoid amending when updating this PR, to make it easier to incrementally review?

**Comment by cmarqu** [comment]
Oh, yes, sure.

**Pull request merged**